api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -410,39 +476,46 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
     if (Strings.isNullOrEmpty(request.getEtag())) {
       throw new BadRequestException("missing required update field 'etag'");
     }
-    long dataSetWorkspaceId =
-        workspaceAuthService
-            .getWorkspaceEnforceAccessLevelAndSetCdrVersion(
-                workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER)
-            .getWorkspaceId();
-    request.setWorkspaceId(dataSetWorkspaceId);
```

Comment:
Sorry, for the confusion. I'm saying that your code changes have created a regression. The DataSetRequest object has a null workspaceId that causes problems downstream when calling save on the dataset. Looks like the the DataSetMapper#dataSetRequestToDb method copies the null workspaceId from DataSetRequest to DbDataset as a 0 since it is a primitive type long, then tries to save the dataset and generates DataIntegrityViolationException

Reply:
My fault - I misunderstood which line you were commenting on. Thanks for catching this - I just repro'd the error locally as well and realized that this line should not have been removed. I got this mixed up with the areas where I left TODO's in this file.